### ISSUE TITLE
Svtplay gives error

### ISSUE THREAD

**Issue by connygunnt** [stefansundin/privatkopiera#40]
```
C:\Users\Conny\Desktop\ffmpeg-4.2.2-win64-static\bin>ffmpeg -i "https://svt-vod-3h.akamaized.net/d0/se/20200917/1965fed1-b16d-450b-9e8a-7e6d32e47b3e/hls-ts-avc.m3u8?alt=https%3A%2F%2Fswitcher.cdn.svt.se%2F1965fed1-b16d-450b-9e8a-7e6d32e47b3e%2Fhls-ts-avc.m3u8" -acodec copy -vcodec copy -absf aac_adtstoasc "'71 - '71.mp4"
ffmpeg version 4.2.2 Copyright (c) 2000-2019 the FFmpeg developers
  built with gcc 9.2.1 (GCC) 20200122
  configuration: --enable-gpl --enable-version3 --enable-sdl2 --enable-fontconfig --enable-gnutls --enable-iconv --enable-libass --enable-libdav1d --enable-libbluray --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libopus --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libtheora --enable-libtwolame --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libzimg --enable-lzma --enable-zlib --enable-gmp --enable-libvidstab --enable-libvorbis --enable-libvo-amrwbenc --enable-libmysofa --enable-libspeex --enable-libxvid --enable-libaom --enable-libmfx --enable-amf --enable-ffnvcodec --enable-cuvid --enable-d3d11va --enable-nvenc --enable-nvdec --enable-dxva2 --enable-avisynth --enable-libopenmpt
  libavutil      56. 31.100 / 56. 31.100
  libavcodec     58. 54.100 / 58. 54.100
  libavformat    58. 29.100 / 58. 29.100
  libavdevice    58.  8.100 / 58.  8.100
  libavfilter     7. 57.100 /  7. 57.100
  libswscale      5.  5.100 /  5.  5.100
  libswresample   3.  5.100 /  3.  5.100
  libpostproc    55.  5.100 / 55.  5.100
[https @ 0000027d1c8fa040] HTTP error 403 Forbidden
https://svt-vod-3h.akamaized.net/d0/se/20200917/1965fed1-b16d-450b-9e8a-7e6d32e47b3e/hls-ts-avc.m3u8?alt=https%3A%2F%2Fswitcher.cdn.svt.se%2F1965fed1-b16d-450b-9e8a-7e6d32e47b3e%2Fhls-ts-avc.m3u8: Server returned 403 Forbidden (access denied)
```

I can watch it via the Proxy (not in Sweden)

**Comment by stefansundin** [comment]
So you are saying that it works via a proxy located in Sweden? But then you can't download without the proxy (i.e. outside of Sweden)?

Some programs/videos are only available to watch inside of Sweden. In that case I think your solution is to use the proxy. :)

If you have another question then please clarify.

**Comment by connygunnt** [comment]
I m using a proxy and can watch all the content , but only download some ,
same goes for TV4 Play

BR Conny

**Comment by Bub0** [comment]
The proxy could be banned, have you tried another or a VPN? I know that OVPN unlock many streamingsites.
If you've scroll down on a episode and click "Mer om programmet" you can see down to the right "rättigheter" ie rigths (where it's avalible to watch)

**Comment by connygunnt** [comment]
SOrry meant VPN , and it is content that is only available in Sweden, as an
example

[image: image.png]

**Comment by Blixtgo** [comment]
I used 0.22 and got this error message. Advice?  Thanks much!

C:\WINDOWS\system32>ffmpeg -i "https://svt-vod-2d.akamaized.net/d0/se/20200926/1b4b074c-9b99-4ba1-ad60-e9deece3fdf1/hls-video-avc-4/hls-video-avc-4.m3u8" -i "https://svt-vod-2d.akamaized.net/d0/se/20200926/1b4b074c-9b99-4ba1-ad60-e9deece3fdf1/hls-audio-stereo-0/hls-audio-stereo-0.m3u8" -acodec copy -vcodec copy -absf aac_adtstoasc "Jacques Brel och hans sånger - Jacques Brel och hans sånger.mp4"
ffmpeg version 4.3.1-essentials_build-www.gyan.dev Copyright (c) 2000-2020 the FFmpeg developers
  built with gcc 10.2.0 (Rev3, Built by MSYS2 project)
  configuration: --enable-gpl --enable-version3 --enable-static --disable-w32threads --disable-autodetect --enable-fontconfig --enable-iconv --enable-gnutls --enable-libxml2 --enable-gmp --enable-lzma --enable-zlib --enable-libsrt --enable-libssh --enable-libzmq --enable-avisynth --enable-sdl2 --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxvid --enable-libaom --enable-libopenjpeg --enable-libvpx --enable-libass --enable-libfreetype --enable-libfribidi --enable-libvidstab --enable-libvmaf --enable-libzimg --enable-amf --enable-cuda-llvm --enable-cuvid --enable-ffnvcodec --enable-nvdec --enable-nvenc --enable-d3d11va --enable-dxva2 --enable-libmfx --enable-libgme --enable-libopenmpt --enable-libopencore-amrwb --enable-libmp3lame --enable-libtheora --enable-libvo-amrwbenc --enable-libgsm --enable-libopencore-amrnb --enable-libopus --enable-libspeex --enable-libvorbis --enable-librubberband
  libavutil      56. 51.100 / 56. 51.100
  libavcodec     58. 91.100 / 58. 91.100
  libavformat    58. 45.100 / 58. 45.100
  libavdevice    58. 10.100 / 58. 10.100
  libavfilter     7. 85.100 /  7. 85.100
  libswscale      5.  7.100 /  5.  7.100
  libswresample   3.  7.100 /  3.  7.100
  libpostproc    55.  7.100 / 55.  7.100
[hls @ 000001f97dacdc80] Skip ('#EXT-X-VERSION:4')
[hls @ 000001f97dacdc80] Opening 'https://svt-vod-2d.akamaized.net/d0/se/20200926/1b4b074c-9b99-4ba1-ad60-e9deece3fdf1/hls-video-avc-4/hls-video-avc-4-1.ts' for reading
[hls @ 000001f97dacdc80] Opening 'https://svt-vod-2d.akamaized.net/d0/se/20200926/1b4b074c-9b99-4ba1-ad60-e9deece3fdf1/hls-video-avc-4/hls-video-avc-4-2.ts' for reading
[https @ 000001f97e140040] Opening 'https://svt-vod-2d.akamaized.net/d0/se/20200926/1b4b074c-9b99-4ba1-ad60-e9deece3fdf1/hls-video-avc-4/hls-video-avc-4-3.ts' for reading
Input #0, hls, from 'https://svt-vod-2d.akamaized.net/d0/se/20200926/1b4b074c-9b99-4ba1-ad60-e9deece3fdf1/hls-video-avc-4/hls-video-avc-4.m3u8':
  Duration: 01:55:57.76, start: 0.100000, bitrate: N/A
  Program 0
    Metadata:
      variant_bitrate : 0
    Stream #0:0: Video: h264 (High) ([27][0][0][0] / 0x001B), yuv420p, 1920x1080 [SAR 1:1 DAR 16:9], 25 fps, 25 tbr, 90k tbn, 50 tbc
    Metadata:
      variant_bitrate : 0
[hls @ 000001f97e454040] Skip ('#EXT-X-VERSION:4')
[hls @ 000001f97e454040] Opening 'https://svt-vod-2d.akamaized.net/d0/se/20200926/1b4b074c-9b99-4ba1-ad60-e9deece3fdf1/hls-audio-stereo-0/hls-audio-stereo-0-1.aac' for reading
[hls @ 000001f97e454040] Opening 'https://svt-vod-2d.akamaized.net/d0/se/20200926/1b4b074c-9b99-4ba1-ad60-e9deece3fdf1/hls-audio-stereo-0/hls-audio-stereo-0-2.aac' for reading
Input #1, hls, from 'https://svt-vod-2d.akamaized.net/d0/se/20200926/1b4b074c-9b99-4ba1-ad60-e9deece3fdf1/hls-audio-stereo-0/hls-audio-stereo-0.m3u8':
  Duration: 01:55:57.80, start: 0.057333, bitrate: N/A
  Program 0
    Metadata:
      variant_bitrate : 0
    Stream #1:0: Audio: aac (LC), 48000 Hz, stereo, fltp, 173 kb/s
    Metadata:
      variant_bitrate : 0
      id3v2_priv.com.apple.streaming.transportStreamTimestamp: \x00\x00\x00\x00\x00\x00\x14(
Jacques Brel och hans sånger - Jacques Brel och hans sånger.mp4: Permission denied

C:\WINDOWS\system32>

**Comment by stefansundin** [comment]
> Jacques Brel och hans sånger - Jacques Brel och hans sånger.mp4: Permission denied
> 
> C:\WINDOWS\system32>

Your user does not have write access to `C:\WINDOWS\system32`. Don't try to run it in `C:\` either, you will get the same error. I suggest that you run the command from your `Desktop` directory.

There is a warning regarding this on the website.. but maybe I need to rephrase it to make it more clear.

**Comment by Blixtgo** [comment]
Aha! I didn’t really understand how to run the command from the desktop. But your mesage got me going and I was able to figure out how to do it. Thanks much. Yes, perhaps you might want to provide an example for those who are not savvy enough to set the directory for the desktop.

Regards

Never put off to tomorrow what you can put off to the day after tomorrow.  -Mark Twain

Från: Stefan Sundin<mailto:notifications@github.com>
Skickat: den 28 september 2020 18:34
Till: stefansundin/privatkopiera<mailto:privatkopiera@noreply.github.com>
Kopia: Blixtgo<mailto:bill.gordon.sweden@hotmail.com>; Comment<mailto:comment@noreply.github.com>
Ämne: Re: [stefansundin/privatkopiera] Svtplay gives error (#40)


Jacques Brel och hans sånger - Jacques Brel och hans sånger.mp4: Permission denied

C:\WINDOWS\system32>

Your user does not have write access to C:\WINDOWS\system32. Don't try to run it in C:\ either, you will get the same error. I suggest that you run the command from your Desktop directory.

There is a warning regarding this on the website.. but maybe I need to rephrase it to make it more clear.

**Comment by stefansundin** [comment]
There is an example [on the website](https://stefansundin.github.io/privatkopiera/#ffmpeg) already (but in Swedish). Can you think of any way to improve it?

<img width="802" alt="Screen Shot 2020-09-28 at 14 45 36" src="https://user-images.githubusercontent.com/1991151/94489514-6b7e5e80-0199-11eb-880f-ff28d7a55c0b.png">

And the link goes to this image:

![](https://stefansundin.github.io/privatkopiera/img/cmd.png)

**Comment by Blixtgo** [comment]
Well, I completely missed that part of your description. Probably because it was under the heading about installing ffmpeg. I never clicked on the link to the screenshot.

If others have had a similar problem, you could move that part and put it in your Följ dessa steg part. Revamp your six steps to include info for dummies. (Like me.) And show your screenshot in the text without requiring me to click. You could also tell how to add your add-on to firefox, which I had never done Before and it took some experimenting to see how.

To sum up, you could make it easier for less experienced users, if you Think of them when you write your Följ dessa steg stuff.

Regards
Bill

Never put off to tomorrow what you can put off to the day after tomorrow.  -Mark Twain

Från: Stefan Sundin<mailto:notifications@github.com>
Skickat: den 28 september 2020 23:48
Till: stefansundin/privatkopiera<mailto:privatkopiera@noreply.github.com>
Kopia: Blixtgo<mailto:bill.gordon.sweden@hotmail.com>; Comment<mailto:comment@noreply.github.com>
Ämne: Re: [stefansundin/privatkopiera] Svtplay gives error (#40)


There is an example on the website<https://stefansundin.github.io/privatkopiera/#ffmpeg> already (but in Swedish). Can you think of any way to improve it?

[Screen Shot 2020-09-28 at 14 45 36]<https://user-images.githubusercontent.com/1991151/94489514-6b7e5e80-0199-11eb-880f-ff28d7a55c0b.png>

And the link goes to this image:

[https://camo.githubusercontent.com/9979469c92795d0eb710ee7cb7adca1a037822a4/68747470733a2f2f73746566616e73756e64696e2e6769746875622e696f2f7072697661746b6f70696572612f696d672f636d642e706e67]<https://camo.githubusercontent.com/9979469c92795d0eb710ee7cb7adca1a037822a4/68747470733a2f2f73746566616e73756e64696e2e6769746875622e696f2f7072697661746b6f70696572612f696d672f636d642e706e67>

**Comment by stefansundin** [comment]
> Well, I completely missed that part of your description. Probably because it was under the heading about installing ffmpeg. I never clicked on the link to the screenshot. If others have had a similar problem, you could move that part and put it in your Följ dessa steg part.

Good idea.

> show your screenshot in the text without requiring me to click.

Hmm.. maybe.

> You could also tell how to add your add-on to firefox.

There is a big blue Firefox button at the top of the webpage. The user's browser is detected and either the Chrome or the Firefox button is made blue depending on what browser the user is running. The button goes to https://addons.mozilla.org/en-US/firefox/addon/privatkopiera/. Hopefully the user has added an extension to Firefox previously and know what to do on this page.
